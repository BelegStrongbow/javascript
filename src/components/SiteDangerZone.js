import React from "react";
import styled from "styled-components";
import CollapsibleHeader from "./CollapsibleHeader";
import { injectIntl, intlShape, defineMessages } from "react-intl";
import { RedButton } from "./Button";
import Paper from "./Paper";

const messages = defineMessages( {
	heading: {
		id: "site_danger_zone.heading",
		defaultMessage: "Danger zone",
	},
	siteRemovalHeading: {
		id: "site_danger_zone.site_removal.heading",
		defaultMessage: "Disconnect this site",
	},
	siteRemovalExplanation: {
		id: "site_danger_zone.site_removal.explanation",
		defaultMessage:	"This will revoke the licenses of the active plugins listed" +
						" above and remove the site from your my.yoast account. This" +
						" will NOT delete your actual website or the plugins" +
						" themselves. You can always re-add the plugins later.",
	},
	siteRemovalButtonText: {
		id: "site_danger_zone.site_removal.button_text",
		defaultMessage: "Remove",
	},
	siteRemovalLoadingButtonText: {
		id: "site_danger_zone.site_removal.loading_button_text",
		defaultMessage: "Removing site...",
	},
} );

const SubHeading = styled.h2`
	font-weight: 600;
	font-size: 1em;
`;

const Container = styled.div`
	padding: 0 40px 16px 40px;

	@media screen and ( max-width: 600px ) {
		padding: 0 20px 16px 20px;
	}
`;

/**
 * Creates Site danger zone element
 *
 * @param {object} props Properties for this element.
 * @returns {ReactElement} SiteDangerZone element.
 * @constructor
 */
function SiteDangerZone( props ) {
	let buttonText = props.intl.formatMessage( messages.siteRemovalButtonText );
	let disabled = {};
	if ( props.removing ) {
		buttonText = props.intl.formatMessage( messages.siteRemovalLoadingButtonText );
		disabled = { disabled };
	}

	return (
		<Paper>
			<CollapsibleHeader title={ props.intl.formatMessage( messages.heading ) } isOpen={ true }>
				<Container>
					<SubHeading>{ props.intl.formatMessage( messages.siteRemovalHeading ) }</SubHeading>
					<p>{ props.intl.formatMessage( messages.siteRemovalExplanation ) }</p>
					<RedButton {...disabled} onClick={ props.onRemove }>{ buttonText }</RedButton>
				</Container>
			</CollapsibleHeader>
		</Paper>
	);
}

SiteDangerZone.propTypes = {
	onRemove: React.PropTypes.func.isRequired,
	removing: React.PropTypes.bool.isRequired,
	intl: intlShape.isRequired,
};

export default injectIntl( SiteDangerZone );
