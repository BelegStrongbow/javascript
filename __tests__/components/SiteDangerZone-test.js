import React from 'react';
import { createComponentWithIntl } from "../../utils";
import SiteDangerZone from '../../src/components/SiteDangerZone';

test('the site dangerzone matches the snapshot', () => {
	const component =  createComponentWithIntl(
		<SiteDangerZone removing={ false } onRemove={ () => {
			console.log( "Remove site" );
		} } />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('the site dangerzone onRemove handling', () => {
	const component =  createComponentWithIntl(
		<SiteDangerZone removing={ false } onRemove={ () => {
			console.log( "Remove site" );
		} } />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

	// manually trigger the callback on the toggle.
	tree.props.onRemove;

	// re-rendering
	tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
