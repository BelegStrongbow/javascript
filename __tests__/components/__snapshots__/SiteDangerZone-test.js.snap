// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the site dangerzone matches the snapshot 1`] = `
<div
  className="dnTGWH"
>
  <div
    className="cxrbkq"
  >
    <button
      aria-expanded={false}
      className="hJrCTB"
      onClick={[Function]}
    >
      <span
        className="ObzGJ"
      >
        <span>
          Disconnect this site
        </span>
        <svg
          fill="#555"
          height={40}
          viewBox="0 0 24 24"
          width={40}
        >
          <path
            d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
          />
        </svg>
      </span>
      <span
        className="GHtNJ"
      />
    </button>
  </div>
</div>
`;

exports[`the site dangerzone onRemove handling 1`] = `
<div
  className="dnTGWH"
>
  <div
    className="cxrbkq"
  >
    <button
      aria-expanded={false}
      className="hJrCTB"
      onClick={[Function]}
    >
      <span
        className="ObzGJ"
      >
        <span>
          Disconnect this site
        </span>
        <svg
          fill="#555"
          height={40}
          viewBox="0 0 24 24"
          width={40}
        >
          <path
            d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
          />
        </svg>
      </span>
      <span
        className="GHtNJ"
      />
    </button>
  </div>
</div>
`;

exports[`the site dangerzone onRemove handling 2`] = `
<div
  className="dnTGWH"
>
  <div
    className="cxrbkq"
  >
    <button
      aria-expanded={false}
      className="hJrCTB"
      onClick={[Function]}
    >
      <span
        className="ObzGJ"
      >
        <span>
          Disconnect this site
        </span>
        <svg
          fill="#555"
          height={40}
          viewBox="0 0 24 24"
          width={40}
        >
          <path
            d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
          />
        </svg>
      </span>
      <span
        className="GHtNJ"
      />
    </button>
  </div>
</div>
`;

exports[`the site dangerzone with removing = true matches the snapshot 1`] = `
<div
  className="dnTGWH"
>
  <div
    className="cxrbkq"
  >
    <button
      aria-expanded={false}
      className="hJrCTB"
      onClick={[Function]}
    >
      <span
        className="ObzGJ"
      >
        <span>
          Disconnect this site
        </span>
        <svg
          fill="#555"
          height={40}
          viewBox="0 0 24 24"
          width={40}
        >
          <path
            d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
          />
        </svg>
      </span>
      <span
        className="GHtNJ"
      />
    </button>
  </div>
</div>
`;
